subprojects/packagefiles/s2n-tls/meson.build

```diff
@@ -0,0 +1,665 @@
+  'tls/s2n_aead.c',
+  'tls/s2n_alerts.c',
+  'tls/s2n_async_pkey.c',
+  'tls/s2n_auth_selection.c',
+  'tls/s2n_cbc.c',
+  'tls/s2n_certificate_keys.c',
+  'tls/s2n_change_cipher_spec.c',
+  'tls/s2n_cipher_preferences.c',
+  'tls/s2n_cipher_suites.c',
+  'tls/s2n_client_cert.c',
+  'tls/s2n_client_cert_verify.c',
+  'tls/s2n_client_finished.c',
+  'tls/s2n_client_hello.c',
+  'tls/s2n_client_hello_request.c',
+  'tls/s2n_client_key_exchange.c',
+  'tls/s2n_config.c',
+  'tls/s2n_connection.c',
+  'tls/s2n_connection_serialize.c',
+  'tls/s2n_crl.c',
+  'tls/s2n_crypto.c',
+  'tls/s2n_early_data.c',
+  'tls/s2n_early_data_io.c',
+  'tls/s2n_ecc_preferences.c',
+  'tls/s2n_encrypted_extensions.c',
+  'tls/s2n_establish_session.c',
+  'tls/s2n_fingerprint.c',
+  'tls/s2n_fingerprint_ja3.c',
+  'tls/s2n_fingerprint_ja4.c',
+  'tls/s2n_handshake.c',
+  'tls/s2n_handshake_hashes.c',
+  'tls/s2n_handshake_io.c',
+  'tls/s2n_handshake_transcript.c',
+  'tls/s2n_handshake_type.c',
+  'tls/s2n_kem.c',
+  'tls/s2n_kem_preferences.c',
+  'tls/s2n_kex.c',
+  'tls/s2n_key_log.c',
+  'tls/s2n_key_update.c',
+  'tls/s2n_ktls.c',
+  'tls/s2n_ktls_io.c',
+  'tls/s2n_next_protocol.c',
+  'tls/s2n_ocsp_stapling.c',
+  'tls/s2n_post_handshake.c',
+  'tls/s2n_prf.c',
+  'tls/s2n_protocol_preferences.c',
+  'tls/s2n_psk.c',
+  'tls/s2n_quic_support.c',
+  'tls/s2n_record_read.c',
+  'tls/s2n_record_read_aead.c',
+  'tls/s2n_record_read_cbc.c',
+  'tls/s2n_record_read_composite.c',
+  'tls/s2n_record_read_stream.c',
+  'tls/s2n_record_write.c',
+  'tls/s2n_recv.c',
+  'tls/s2n_renegotiate.c',
+  'tls/s2n_resume.c',
+  'tls/s2n_security_policies.c',
+  'tls/s2n_security_rules.c',
+  'tls/s2n_send.c',
+  'tls/s2n_server_cert.c',
+  'tls/s2n_server_cert_request.c',
+  'tls/s2n_server_done.c',
+  'tls/s2n_server_extensions.c',
+  'tls/s2n_server_finished.c',
+  'tls/s2n_server_hello.c',
+  'tls/s2n_server_hello_retry.c',
+  'tls/s2n_server_key_exchange.c',
+  'tls/s2n_server_new_session_ticket.c',
+  'tls/s2n_shutdown.c',
+  'tls/s2n_signature_algorithms.c',
+  'tls/s2n_signature_scheme.c',
+  'tls/s2n_tls.c',
+  'tls/s2n_tls13.c',
+  'tls/s2n_tls13_certificate_verify.c',
+  'tls/s2n_tls13_handshake.c',
+  'tls/s2n_tls13_key_schedule.c',
+  'tls/s2n_tls13_secrets.c',
+  'tls/s2n_x509_validator.c',
+  'utils/s2n_array.c',
+  'utils/s2n_atomic.c',
+  'utils/s2n_blob.c',
+  'utils/s2n_ensure.c',
+  'utils/s2n_fork_detection.c',
+  'utils/s2n_init.c',
+  'utils/s2n_io.c',
+  'utils/s2n_map.c',
+  'utils/s2n_mem.c',
+  'utils/s2n_random.c',
+  'utils/s2n_rfc5952.c',
+  'utils/s2n_safety.c',
+  'utils/s2n_socket.c',
+  'utils/s2n_timer.c',
+]
+
+libs2n = library(
+  's2n',
+  src,
+  include_directories: [private_include, public_include],
+  dependencies: deps,
+  c_args: c_args,
+  install: true,
+  version: meson.project_version(),
+)
+
+s2n_tls_dep = declare_dependency(
+  include_directories: [public_include],
+  link_with: libs2n,
+  dependencies: deps,
```

Comment:
Minor: assuming s2n's users don't directly use its dependencies, this line can be dropped.